src/runtime-rs/src/shim/src/task_service.rs

```diff
@@ -0,0 +1,848 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use ::log::{debug, error, info, trace, warn};
+
+use ttrpc::error::get_rpc_status;
+use ttrpc::{Code, Result, TtrpcContext};
+use virtcontainers::Sandbox;
+
+use shim_proto::task::{ProcessInfo, Status};
+use shim_proto::{empty::Empty, metrics, shim, shim_ttrpc::Task};
+use std::convert::{TryFrom, TryInto};
+
+use protobuf::Message;
+
+use agent::{BlkioStatsEntry, StatsContainerResponse};
+use nix::sys::signal::Signal;
+use std::sync::mpsc::Sender;
+use std::sync::{Arc, Mutex};
+use virtcontainers::container::State;
+use virtcontainers::oci::ContainerType;
+
+use crate::run::{ServerAction, ServerMessage};
+
+#[allow(dead_code)]
+pub(crate) struct TaskService {
+    pub id: String,
+    pub pid: u32,
+    pub sandbox: Arc<Mutex<Sandbox>>,
+    pub shutdown_sender: Mutex<Sender<ServerMessage>>,
+}
+
+impl Task for TaskService {
+    fn state(&self, _ctx: &TtrpcContext, req: shim::StateRequest) -> Result<shim::StateResponse> {
+        debug!("====> state: {:?}", req);
+        let rsp = self.do_state(req);
+        debug!("<==== state: {:?}", rsp);
```

Comment:
I'm not sure we need all these debug calls. It also relates to a missing feature: tracing, which would show all the function  calls and exits automatically if you decorated these methods with `#[instrument]`.

Reply:
Leave it to the later revisions. Using tracing is a better choice.

Reply:
I have tried `#[instrument]` , but it only create a span that include input arguments. I think it is more for tracing, but not  automatically recording input and output values. So I keep these debug calls unchanged to record more information in debug mode.